prov/efa/src/efa_domain.c

```diff
@@ -236,22 +239,27 @@ int efa_domain_open(struct fid_fabric *fabric_fid, struct fi_info *info,
 	 * If FI_MR_LOCAL is set, we do not want to use the
 	 * MR cache
 	 */
-	if (info->domain_attr && info->domain_attr->mr_mode & FI_MR_LOCAL) {
-		efa_mr_cache_enable = 0;
+	if (efa_mr_cache_enable && info->domain_attr && info->domain_attr->mr_mode & ~FI_MR_LOCAL) {
+		domain->cache = (struct ofi_mr_cache *)calloc(1, sizeof(struct ofi_mr_cache));
+		if (!domain->cache) {
+			ret = -FI_ENOMEM;
+			goto err_free_info;
+		}
 	}
```

Comment:
I suspect calloc is being called here but I'd be explicit anyway and set domain->cache to NULL when not enabled.

Reply:
Yeah I can do that